### PR TITLE
Sync java and go sidecar clients

### PR DESCRIPTION
**Pull request by alourie**
WIP

Also has a start for the backups.

Signed-off-by: Alex Lourie <djay.il@gmail.com>

### CODE REVIEW

File: pkg/controller/cassandradatacenter/statefulset.go [hunk 1]
```diff
@@ -398,6 +381,21 @@ func podsInIncorrectCassandraState(desiredReplicas int32, existingReplicas int32
 	return podsByIncorrectCassandraMode
 }
 
+func setControllerReference(
```
Comment (alourie) [inline pkg/controller/cassandradatacenter/statefulset.go]: will remove this, it's not needed

File: pkg/sidecar/client.go [hunk 2]
```diff
@@ -14,13 +15,16 @@ import (
...
-	applicationJson    = "application/json"
 	EndpointOperations = "operations"
 	EndpointStatus     = "status"
 )
 
-var DefaultSidecarClientOptions = ClientOptions{Port: 4567, Secure: false,}
+var (
+	accept          = http.CanonicalHeaderKey("Accept")
+	applicationJson = "application/json; charset=utf-8"
+)
```
Comment (alourie) [inline pkg/sidecar/client.go]: will restore this to const

File: pkg/sidecar/client.go [hunk 3]
```diff
@@ -82,13 +86,14 @@ func NewSidecarClient(host string, options *ClientOptions) *Client {
...
-func SidecarClients(pods []corev1.Pod, clientOptions ClientOptions) map[*corev1.Pod]*Client {
+func SidecarClients(pods []corev1.Pod, clientOptions *ClientOptions) map[*corev1.Pod]*Client {
 
 	podClients := make(map[*corev1.Pod]*Client)
 
-	for _, pod := range pods {
-		key := pod
-		podClients[&key] = NewSidecarClient(pod.Status.PodIP, &clientOptions)
+	for i, pod := range pods {
+		if pod.Status.Phase == corev1.PodRunning {
```
Comment (alourie) [inline pkg/sidecar/client.go]: keep this `if` or remove?

**Comment by smiklosovic** [comment]
looks good to me

File: pkg/controller/cassandrabackup/reconciler.go [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+// TODO(user): Modify this Reconcile function to implement your Controller logic.  This example creates
+// a Pod as an example
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileCassandraBackup) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling CassandraBackup")
+
+	// backup and shit
```
Comment (smiklosovic) [inline pkg/controller/cassandrabackup/reconciler.go]: i would omit any explicit language in Instaclustr projects

Comment (alourie) [reply pkg/controller/cassandrabackup/reconciler.go]: sure ;-)

**Pull request merged**